ml-agents/mlagents/trainers/ppo/policy.py

```diff
@@ -111,13 +135,19 @@ def update(self, mini_batch, num_sequences):
             self.model.batch_size: num_sequences,
             self.model.sequence_length: self.sequence_length,
             self.model.mask_input: mini_batch["masks"].flatten(),
-            self.model.returns_holder: mini_batch["discounted_returns"].flatten(),
-            self.model.old_value: mini_batch["value_estimates"].flatten(),
             self.model.advantage: mini_batch["advantages"].reshape([-1, 1]),
             self.model.all_old_log_probs: mini_batch["action_probs"].reshape(
                 [-1, sum(self.model.act_size)]
             ),
         }
+        for i, name in enumerate(self.reward_signals.keys()):
+            feed_dict[self.model.returns_holders[i]] = mini_batch[
```

Comment:
Are the `returns_holders` and `old_values` guaranteed to be in same order as reward_signals?  

Can we just key them by the name instead of in a list? (this goes back to my question about reward_signals vs. stream_names)

Reply:
You're exactly right - on my other branch I've changed it so that the model only refers to them by name and not index. I think I'll make these changes here as well

Reply:
I am now referring these by name - marking as resolved